openslides_backend/action/motion_category/number_motions.py

```diff
@@ -0,0 +1,225 @@
+from typing import Dict, List, Optional
+
+from ...models.models import Motion, MotionCategory
+from ...services.datastore.commands import GetManyRequest
+from ...shared.exceptions import ActionException
+from ...shared.patterns import Collection, FullQualifiedId
+from ..base import ActionPayload
+from ..default_schema import DefaultSchema
+from ..generics import UpdateAction
+from ..register import register_action
+
+
+@register_action("motion_category.number_motions")
+class MotionCategoryNumberMotions(UpdateAction):
+    """
+    Action to number motions for a category.
+    """
+
+    model = Motion()
+    schema = DefaultSchema(MotionCategory()).get_default_schema(
+        title="Number motions of a motion category.",
+        description="An object containing an array of main category id.",
+        required_properties=["id"],
+    )
+
+    def get_updated_instances(self, payload: ActionPayload) -> ActionPayload:
+        for instance in payload:
+            self.init_memory(instance["id"])
+
+            affected_categories = self.get_affected_categories(instance["id"])
+            affected_motions = self.get_affected_motions(affected_categories)
+            non_affected_motions = [
+                id_ for id_ in self.mem_motions if id_ not in affected_motions
+            ]
+            existing_numbers = [
+                self.mem_motions[id_].get("number") for id_ in non_affected_motions
+            ]
+
+            # check for missing lead_motion_ids in affected_motions.
+            for motion_id in affected_motions:
+                lead_motion_id = self.get_lead_motion_id(motion_id)
+                if lead_motion_id and lead_motion_id not in affected_motions:
+                    raise ActionException(
+                        f'Amendment "{motion_id}" cannot be numbered, because it\'s lead motion ({lead_motion_id}) is not in category {instance["id"]} or any subcategory.'
+                    )
+
+            # generate number_value_map with the number_values for motion_ids.
+            main_counter = 1
+            lead_motions_counter: Dict[int, int] = dict()
```

Comment:
```suggestion
            lead_motions_counter: Dict[int, int] = defaultdict(lambda: 1)
```
with `from collections import defaultdict`. This saves the `else` part, if a lead motion is not in the counter

Reply:
Done.